### ISSUE TITLE
Error: font can't be updated

### ISSUE THREAD

**Issue by xbladesub** [neovide/neovide#2340]
How to fix this? Update from 11.1 to 12.2 and got this message on startup. Neovim version -  0.9.2

Also `:q` doesn't work and entire neovide instance freezes..

MacOS

<img width="1170" alt="Screenshot 2024-02-04 at 8 11 41 PM" src="https://github.com/neovide/neovide/assets/45129790/fe67a85b-f390-4ce7-9f25-4e8baaf4ce8c">

**Comment by fredizzimo** [comment]
I don't know the cause of the freeze, that's one thing that was a regression in 0.12.1 and fixed in 0.12.2. So a log would be great. 

But for the font. Neovide does not support specifying families as the font names (at least not on all platforms, it might work on some), so `monospace` is not valid. It's also possible that you have mistyped the name and it should be [mon**a**space](https://monaspace.githubnext.com/) 

As a side-note, we should improve the error message too, it's too technical, the full details can be included but it should tell the primary cause first.

**Comment by xbladesub** [comment]
> I don't know the cause of the freeze, that's one thing that was a regression in 0.12.1 and fixed in 0.12.2. So a log would be great.
> 
> But for the font. Neovide does not support specifying families as the font names (at least not on all platforms, it might work on some), so `monospace` is not valid. It's also possible that you have mistyped the name and it should be [mon**a**space](https://monaspace.githubnext.com/)
> 
> As a side-note, we should improve the error message too, it's too technical, the full details can be included but it should tell the primary cause first.

```lua
	if is_mac_mini then
		o.guifont = "JetBrainsMono Nerd Font:h15"
	else
		o.guifont = "JetBrainsMono Nerd Font:h11"
	end
```
This is everyting in my config which is related to a font

**Comment by fredizzimo** [comment]
In that case check that you don't have a `ginit.vim` file, that file is also read by Neovide.

**Comment by dfendr** [comment]
I fixed this error for myself on MacOS 14.3

This code snippet was causing issues.

```lua
    vim.opt[guifont] = "monospace:h17"
```

Commenting it out fixed the error popup.

**Comment by Paliago** [comment]
I'm also on MacOS 14.2 and getting this issue.
I'm using Lunarvim as the bin to launch and it sets the default guifont to "monospace:h15".
I've changed it to my font of choice in my config and running `:set guifont` it echos back the correct one. Still the error shows up during launch of neovide.

**Comment by fredizzimo** [comment]
Can someone report this bug to LunarVim? It should not set the guifont, especially not to something that is not available on every system. 

It's up to the Gui to select a suitable monospace font if none is given.

Neovide loads the font each time it's set, so that's why it gives errors even if you later set it yourself.

**Comment by fredizzimo** [comment]
Sorry, I should have checked, it's already reported
* https://github.com/LunarVim/LunarVim/issues/4446

**Comment by Achim63** [comment]
Same here on macOS 26.3.1 after updating nvim from latest 0.11 to 0.12-dev. 
I still use my old .vimrc and the font is set correctly after hitting return to get rid of the error message:
" neovide options
if has('gui_running') || exists('neovide')
  set guifont=FiraCode\ Nerd\ Font\ Mono:h19
  " some more unrelated neovide options here ...
end

**Comment by falcucci** [comment]
@Achim63 the error probably is due to a neovim upstream change. If you want to use *neovim* nightly you also need to use *neovide* nightly which contains the fix.

**Comment by ecbaldwin** [comment]
I got this error today after upgrading nvim to the released `NVIM v0.12.0`. I was using neovide `0.15.2`. This issue hinted to me that upgrading neovide to the nightly would get me past the error, which it did.

I don't use `LunarVim` at all and I tried to start neovide after removing all local guifont settings. The error still occurred. Only updating to the nightly seemed to fix it for me.

So, I guess now we have a released nvim that requires an unreleased neovide.
